### DISCUSSION TITLE
Digital Ocean -- image from marketplace, DNS issue

### DISCUSSION THREAD

**Discussion by shaib** [nextcloud/all-in-one#3524] [Questions] [answered]
Hi,

I installed a new droplet on Digital Ocean with the marketplace image. The image did not work -- the mastercontainer could not resolve the name "nextcloud.com", and as a result, it could not start, and was stuck in a restart loop. I couldn't even open a shell in it -- I could only see the error messages using `docker logs`.

When I checked `/etc/resolv.conf` on the host, I found it was pointing to 127.0.0.53, and if that is what the container was trying to use, then DNS failures make perfect sense (in the container, 127.0.0.53 is the container, not the host).

I followed https://dockerlabs.collabnix.com/intermediate/networking/Configuring_DNS.html and added DNS configuration to `etc/docker/daemon.json` (except I used 1.1.1.1, not 114.114.114.114 which I'd never heard of). But even stopping and restarting the container didn't change anything -- I had to remove the container and start a new one. I found the command that started the original container in `/var/log/auth.log` (because, although the only user in the system is root, that one was run with sudo), and so I was able to repeat it exactly (I couldn't find it documented anywhere).

```
/usr/bin/docker run -d --name nextcloud-aio-mastercontainer --restart always -p 80:80 -p 8080:8080 -p 8443:8443 -e NEXTCLOUD_MOUNT=/mnt/ -e NEXTCLOUD_DATADIR=/mnt/ncdata --volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config --volume /var/run/docker.sock:/var/run/docker.sock:ro nextcloud/all-in-one:latest
```
Then, everything started to work.

Is this a bug? A Digital-ocean only problem?

**Comment by szaimen** [comment]
Hi, can you check if you might be running into this?
https://github.com/nextcloud/all-in-one/discussions/2065

**Reply by shaib** [reply]
```
# docker network inspect nextcloud-aio | grep Internal
        "Internal": false,
```
I think I had run into that discussion during the troubleshooting, but forgot to mention it. I did not need to recreate the network, only to add the DNS definition.

**Comment by szaimen** [comment, marked as answer]
Hi, I just spun up a new droplet on digital ocean and apparently everything worked - so there was no dns resolution issue on this new test instance. Not sure why you are running into this problem. Did you test with another droplet maybe in another location?

**Reply by shaib** [reply]
Interesting. Can you share `/etc/resolv.conf` from the host?

I'm trying to figure out how name resolution in the docker works. It seems not to be defined by anything in AIO, maybe in alpine?

**Reply by shaib** [reply]
Eh, no, I didn't try with another one.

**Reply by shaib** [reply]
On the way to moving to beta (for #3525), I thought that if I'm already going to replace the master container, I'll make another experiment here. So I moved the `daemon.json` away, restarted the docker service, and recreated the master container (still from latest). This time, it got the Digital Ocean DNS servers (as described [here](https://docs.digitalocean.com/support/check-your-droplets-network-configuration/#correcting-dns-resolution), as one might expect from DHCP. And of course it worked.

So I'm guessing this issue was some kind of fluke, perhaps a temporary DHCP malfunction.
